flang/runtime/ISO_Fortran_binding.cpp

```diff
@@ -75,7 +75,14 @@ RT_API_ATTRS int CFI_allocate(CFI_cdesc_t *descriptor,
     dim->sm = byteSize;
     byteSize *= extent;
   }
+#ifdef _AIX
```

Comment:
I think we could just do 
```
  void *p = byteSize ? std::malloc(byteSize) : nullptr;
```
Because the next check would already issue CFI_ERROR_MEM_ALLOCATION if `byteSize` is zero.

Reply:
+1, and I think you should do that regardless of the platform since std::malloc(0) is undefined.

This is also done like that for the allocations from Fortran since: https://github.com/llvm/llvm-project/pull/66124